src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -283,8 +313,6 @@ def begin_string(self):
         '''
         result = "omp single"
 
-        if self._nowait:
-            result = result + " nowait"
```

Comment:
Won't removing that disable nowait in the LFRic API that still uses gen_code?

Reply:
I think the `gen_code` call still does this, it shoudln't need to be in begin_string right?

Reply:
Ah I see. Can you add the "# pylint: disable=no-self-use" to get rid of the pylint issue. I am right it thinking that in the long term (when all clauses have their class) this could be a static method or simply a string attribute of the class.